dashboard/app/models/plc/course_unit.rb

```diff
@@ -50,4 +50,40 @@ def get_top_modules_of_each_type_from_user_selections(learning_module_ids_and_we
 
     selected_learning_modules
   end
+
+  def determine_preferred_learning_modules(user)
+    evaluation_level = script.levels.reverse.find {|level| level.class == LevelGroup}
+
+    level_source = UserLevel.find_by(user: user, level: evaluation_level, script: script).try(:level_source)
+    return [] if level_source.nil?
+
+    responses = JSON.parse(level_source.data)
+    learning_module_weights = Hash.new(0)
+
+    responses.each do |level_id, response|
+      level = EvaluationMulti.find(level_id)
+      selected_answer = level.answers[response['result'].to_i]
+
+      next if selected_answer['stage'].nil?
+
+      stage = Stage.find_by(name: selected_answer['stage'], script: script)
+      learning_module = stage.try(:plc_learning_module)
+
+      next if learning_module.nil?
+
+      learning_module_weights[learning_module] += selected_answer['weight']
+    end
+
+    learning_module_weights = learning_module_weights.sort_by{|_, weight| weight}
+
+    default_module_assignments = []
+
+    Plc::LearningModule::NONREQUIRED_MODULE_TYPES.each do |module_type|
+      module_to_assign = learning_module_weights.find {|learning_module| learning_module[0].module_type == module_type}.try(:[], 0)
```

Comment:
Nit: if learning_module_weights is an array of pairs you can splat the pairs instead of doing `learning_module[0]`, like this:
`[[0, 4], [1, 1], [3, 4]].find{|a, b| a == b}`
alternatively since you are done with the weight value after sorting it above you could do something like
````
sorted_learning_modules = learning_module_weights.map(&:first)
module_to_assign = sorted_learning_modules.find {|learning_module| learning_module.module_type == module_type}
```
to avoid both the [0] and .try(:[], 0)